import_obfuscation.go

```diff
@@ -0,0 +1,473 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"sort"
+	"strings"
+
+	"github.com/Binject/debug/goobj2"
+)
+
+type pkgInfo struct {
+	pkg  *goobj2.Package
+	path string
+}
+
+type dataType uint8
+
+const (
+	other dataType = iota
+	importPath
+	namedata
+)
+
+func obfuscateImports(objPath, importCfgPath string) (map[string]string, error) {
+	importCfg, err := goobj2.ParseImportCfg(importCfgPath)
+	if err != nil {
+		return nil, err
+	}
+	mainPkg, err := goobj2.Parse(objPath, "main", importCfg)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing main objfile: %v", err)
+	}
+	privatePkgs := []pkgInfo{{mainPkg, objPath}}
+
+	// build list of imported packages that are private
+	for pkgPath, info := range importCfg {
+		if isPrivate(pkgPath) {
+			pkg, err := goobj2.Parse(info.Path, pkgPath, importCfg)
+			if err != nil {
+				return nil, fmt.Errorf("error parsing objfile %s at %s: %v", pkgPath, info.Path, err)
+			}
+
+			privatePkgs = append(privatePkgs, pkgInfo{pkg, info.Path})
+		}
+	}
+
+	var sb strings.Builder
+	var buf bytes.Buffer
```

Comment:
`sb` and `buf` have a scope spanning over 100 lines - they need better names, like what they're actually used for

Reply:
even better; `buf` isn't used until the end of the func, so move the declaration there.

Reply:
I agree I may want to rename them, but they are actually used quite frequently throughout `obfuscateImports` as an optimization of sorts. Because there is the potential to modify hundreds of symbol names/data, they are passed into `garbleSymData` and `garbleSymbolName` so that the Builder/Buffer can be used to efficiently build `[]byte`s or `strings`.